api/all/src/main/java/io/opentelemetry/api/metrics/DoubleHistogramBuilder.java

```diff
@@ -32,6 +34,11 @@ public interface DoubleHistogramBuilder {
    */
   DoubleHistogramBuilder setUnit(String unit);
 
+  /** Specify advice for histogram implementations. */
+  default DoubleHistogramBuilder setAdvice(Consumer<HistogramAdviceConfigurer> adviceConsumer) {
```

Comment:
Nit: are we planning to add this to the API right away? Or perhaps introduce it in an `internal` interface first?

Reply:
I'm open to different options. Could make it an internal interface, or add it to the [api incubator](https://github.com/open-telemetry/opentelemetry-java/tree/main/extensions/incubator), or wait on merging until the spec is marked stable. 

I think I like the idea of an internal interface the best.

Reply:
I've split out the interfaces to `:extensions:incubator`.

To use, add a dependency on `opentelemetry-extension-incubator`, and cast the histogram builder to the extended version of the interface:
-`((ExtendedDoubleHistogramBuilder) meter.histogramBuilder("histogram")).setAdvice(...)`
-`((ExtendedLongHistogramBuilder) meter.histogramBuilder("histogram").ofLongs()).setAdvice(...)`